test/e2e/runner/load.go

```diff
@@ -172,10 +160,21 @@ func createTxBatch(ctx context.Context, txCh chan<- types.Tx, testnet *e2e.Testn
 					panic(fmt.Sprintf("Failed to generate tx: %v", err))
 				}
 
-				select {
-				case txCh <- tx:
-				case <-ctx.Done():
-					return
+				var nodeIndices []int
+				if testnet.LoadNumNodesPerTx <= 1 {
+					// Pick one random node to send the transaction.
+					nodeIndices = []int{cmtrand.Intn(len(testnet.Nodes))}
+				} else {
+					// Pick LoadNumNodesPerTx random nodes (channel indices) to
+					// send the transaction.
+					nodeIndices = cmtrand.Perm(len(testnet.Nodes))[:testnet.LoadNumNodesPerTx]
+				}
+				for _, i := range nodeIndices {
+					select {
+					case txChs[i] <- tx:
```

Comment:
How do we know that we have not picked an index of a node we are skipping ,thus we will not use this channel in the `Load` function? What we can do is keep a list of valid indexes in `Load` passing it to `LoadGenerate` that will forward it here.

Reply:
Good catch. I fixed it.